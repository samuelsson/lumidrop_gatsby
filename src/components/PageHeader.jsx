import React from 'react';
import styled from 'styled-components';

const Header = styled.div`
  padding: 15px;
`;

const PageHeader = () => (
  <Header>
    Header
  </Header>
);

export default PageHeader;
