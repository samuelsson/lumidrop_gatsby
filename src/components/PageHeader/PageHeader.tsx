import React from 'react';
import styled from 'styled-components';
import { colors, variables } from '../../styles';
import Navigation from '../Navigation/Navigation';

const StyledHeaderContainer = styled.header`
  position: relative;
  padding: ${variables.defaultPadding};
  border-bottom: 2px solid ${colors.gray[800]};
  background-color: ${colors.gray[900]};
`;

const StyledHeader = styled.div`
  display: flex;
  margin: 0 auto;
  width: ${variables.breakpoints.large};
`;

const StyledLogoContainer = styled.div`
  position: absolute;
  left: ${variables.defaultPadding};
  margin-right: ${variables.defaultPadding};
`;

const StyledLoginInfo = styled.div`
  position: absolute;
  right: ${variables.defaultPadding};
  margin-left: ${variables.defaultPadding};
`;

const PageHeader = (): JSX.Element => (
  <StyledHeaderContainer>
    <StyledHeader>
      <StyledLogoContainer>Header</StyledLogoContainer>
      <Navigation />
      <StyledLoginInfo>Login info</StyledLoginInfo>
    </StyledHeader>
  </StyledHeaderContainer>
);

export default PageHeader;
