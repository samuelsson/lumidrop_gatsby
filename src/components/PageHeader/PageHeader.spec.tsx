import React from 'react';
import { render, cleanup } from '@testing-library/react';
import PageHeader from './PageHeader';

describe('<PageHeader />', (): void => {
  afterEach(cleanup);

  it('renders the title', (): void => {
    const title = 'Header';
    const { getByText } = render(<PageHeader />);
    expect(getByText(title).tagName).toBeTruthy();
  });

  it('matches snapshot', (): void => {
    const { asFragment } = render(<PageHeader />);
    expect(asFragment()).toMatchSnapshot();
  });
});
