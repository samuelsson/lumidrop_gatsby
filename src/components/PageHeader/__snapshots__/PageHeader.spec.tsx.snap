// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PageHeader /> matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="sc-bdVaJa dgUUwq"
  >
    Header
  </div>
</DocumentFragment>
`;
